### ISSUE TITLE
Parsing error: escaped quotes in string under a group

### ISSUE THREAD

**Issue by paulholden2** [project-eutopia/keisan#105]
Hello, I've run into another issue with escaped quotes in strings. Parsing works great with escaped strings, but if the string literal is enclosed in a group, the tokenizer fails:

```rb
# Simple string
Keisan::Calculator.new.ast('"a"')
# => #<Keisan::AST::String:0x00007fdf97128bc0 @content="a">

# With escaped double quote
Keisan::Calculator.new.ast('"\"a"')
# => #<Keisan::AST::String:0x00007fdf9b8d8c68 @content="\"a">

# Escaped double quote in a group
Keisan::Calculator.new.ast('("\"a")')
# Raises Keisan::Exceptions::TokenizingError (Tokenizing error, no closing quote ")
```

It occurs for any form of group, and either quote style—single or double.

Edit: I should add a more practical example along with an example of some of the more wacky behavior:

```rb
# Calling a function with a string arg that contains an escaped quote
Keisan::Calculator.new.ast('my_fn("\"a")')
# Raises Keisan::Exceptions::TokenizingError (Tokenizing error, no closing quote ")

# The expression gets modified in a strange way if your escaped quotes are actually closed, e.g.
Keisan::Calculator.new.ast('("\"a\"")').to_s
# => "(\"\"*a)*\"\""
# The above line is output from Rails console, so it's an `inspect`ed string. The actual value of the string it spits out is:
# (""*a)*""
# but it should be
# "\"a\""
```

Currently using Keisan version 0.8.4

**Comment by project-eutopia** [comment]
Thanks for finding this, it is definitely a bug. I'll update this thread when I have taken a deep look at the problem.

**Comment by project-eutopia** [comment]
Fixed this in #106, and pushed a version 0.8.5 after merging #107. I checked locally on the new version downloaded from RubyGems and I think it works now, but please check and let me know if this works for you.
